repose-aggregator/tests/functional-tests/src/integrationTest/groovy/features/filters/samlpolicy/SamlAttributeMappingTest.groovy

```diff
@@ -0,0 +1,429 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_-_=_
+ */
+
+package features.filters.samlpolicy
+
+import features.filters.samlpolicy.util.SamlUtilities
+import framework.ReposeValveTest
+import framework.mocks.MockIdentityV2Service
+import groovy.json.JsonSlurper
+import org.opensaml.saml.saml2.core.Attribute
+import org.opensaml.saml.saml2.core.Response as SamlResponse
+import org.rackspace.deproxy.Deproxy
+import org.rackspace.deproxy.Request
+import org.rackspace.deproxy.Response
+import org.spockframework.runtime.ConditionNotSatisfiedError
+import spock.lang.FailsWith
+
+import static features.filters.samlpolicy.util.SamlPayloads.*
+import static features.filters.samlpolicy.util.SamlUtilities.*
+import static framework.mocks.MockIdentityV2Service.createIdpJsonWithValues
+import static framework.mocks.MockIdentityV2Service.createMappingJsonWithValues
+import static javax.servlet.http.HttpServletResponse.SC_OK
+
+/**
+ * This functional test ensures the attribute mappings are being set correctly on the request and the response.
+ */
+class SamlAttributeMappingTest extends ReposeValveTest {
+    static samlUtilities = new SamlUtilities()
+    static xmlSlurper = new XmlSlurper()
+    static jsonSlurper = new JsonSlurper()
+    static MockIdentityV2Service fakeIdentityV2Service
+
+    def setupSpec() {
+        reposeLogSearch.cleanLog()
+
+        def params = properties.defaultTemplateParams
+        repose.configurationProvider.applyConfigs("common", params)
+        repose.configurationProvider.applyConfigs("features/filters/samlpolicy", params)
+
+        deproxy = new Deproxy()
+
+        fakeIdentityV2Service = new MockIdentityV2Service(properties.identityPort, properties.targetPort)
+        deproxy.addEndpoint(properties.targetPort, 'origin service', null, fakeIdentityV2Service.handler)
+        deproxy.addEndpoint(properties.identityPort, 'identity service', null, fakeIdentityV2Service.handler)
+
+        repose.start()
+        reposeLogSearch.awaitByString("Repose ready", 1, 30)
+
+        fakeIdentityV2Service.admin_token = UUID.randomUUID().toString()
+    }
+
+    def setup() {
+        fakeIdentityV2Service.resetHandlers()
+    }
+
+    @FailsWith(ConditionNotSatisfiedError)
+    def "the saml:response will be translated before being sent to the origin service"() {
+        given: "a mapping policy with a literal value and a path-based value in addition to the standard attributes"
+        def extAttribLiteral = "banana"
+        def extAttribLiteralValue = "phone"
+        def extAttribPath = "adventure"
+        def extAttribPathValue = "Mordor"
+        def mappingPolicy = createMappingJsonWithValues(
+                userExtAttribs: [(extAttribLiteral): extAttribLiteralValue, (extAttribPath): "{0}"],
```

Comment:
Having to escape the keys seems so silly. All just so that string keys can be used without quotations. But why would you even want that? The quotations provide a consistent indicator that something is a string. Furthermore, it seems just as likely that you would use a variable for the key as a string, so why prefer the string? This is one of those times where I'm like, man, I'm just not a big fan of Groovy.